build_tools/cmake/iree_check_test.cmake

```diff
@@ -162,6 +162,9 @@ function(iree_check_test)
   if(_RULE_INPUT_TYPE)
     list(APPEND _BASE_COMPILER_FLAGS "--iree-input-type=${_RULE_INPUT_TYPE}")
   endif()
+  if(_RULE_TARGET_CPU)
+    list(APPEND _BASE_COMPILER_FLAGS "--iree-llvmcpu-target-cpu=${_RULE_TARGET_CPU}")
+  endif()
   if(_RULE_TARGET_CPU_FEATURES)
     list(APPEND _BASE_COMPILER_FLAGS "--iree-llvmcpu-target-cpu-features=${_RULE_TARGET_CPU_FEATURES}")
   endif()
```

Comment:
What's the rationale for these getting their own flags instead of using `COMPILER_FLAGS`?

In general, I want these build system functions to do _less_, not _more_ - especially for global flags controlling target backend/device-specific options.

I don't really mind being this explicit in our tests:
```cmake
  TARGET_BACKEND
    "llvm-cpu"
  COMPILER_FLAGS
    "--iree-llvmcpu-target-cpu=generic"
  DRIVER
    "local-task"
```

Reply:
Good catch! I just made that change in all related rules.